client.go

```diff
@@ -143,6 +143,91 @@ func (c *client) createSP(
 	return &result, password, err
 }
 
+// addAppPassword adds a new password to an App's credentials list.
+func (c *client) addAppPassword(ctx context.Context, appObjID string, duration time.Duration) (string, string, error) {
+	keyID, err := uuid.GenerateUUID()
+	if err != nil {
+		return "", "", err
+	}
+
+	// Key IDs are not secret, and they're a convenient way for an operator to identify Vault-generated
+	// passwords. These must be UUIDs, so the three leading bytes will be used as an indicator.
+	keyID = "ffffff" + keyID[6:]
+
+	password, err := uuid.GenerateUUID()
+	if err != nil {
+		return "", "", err
+	}
+
+	now := time.Now().UTC()
+	cred := graphrbac.PasswordCredential{
+		StartDate: &date.Time{Time: now},
+		EndDate:   &date.Time{Time: now.Add(duration)},
+		KeyID:     to.StringPtr(keyID),
+		Value:     to.StringPtr(password),
+	}
+
+	// Load current credentials
+	resp, err := c.provider.ListApplicationPasswordCredentials(ctx, appObjID)
+	if err != nil {
+		return "", "", errwrap.Wrapf("error fetching credentials: {{err}}", err)
+	}
+	curCreds := *resp.Value
+
+	// Add and save credentials
+	curCreds = append(curCreds, cred)
+
+	if _, err := c.provider.UpdateApplicationPasswordCredentials(ctx, appObjID,
+		graphrbac.PasswordCredentialsUpdateParameters{
+			Value: &curCreds,
+		},
+	); err != nil {
+		if strings.Contains(err.Error(), "size of the object has exceeded its limit") {
```

Comment:
This is nice to handle this, even if it eventually breaks due to response changes.